### PR TITLE
(feat)[security-services] Update Kong version to 2.0.1

### LINKED ISSUE #191: Version bump of kong for Geneva
Docker version bump of kong for Geneva

Cloned from https://github.com/edgexfoundry/edgex-go/issues/1935.

Bump kong version in docker-compose.
Snaps must be similarly updated.

Version bump is needed to pick up security fixes and reduce technical debt from using out-of-date container versions.

Blackbox regressions must be run against new version.

### PR DESCRIPTION
**Pull request by bnevis-i**
Fixes #191 

Signed-off-by: Bryon Nevis <bryon.nevis@intel.com>

### CODE REVIEW

**Comment by bnevis-i** [comment]
Blackbox validation results:

-co: 1 failure (same as baseline)
-dv: 137 failures (same as baseline)
-sec: unable to run due to script issue

cd, md, sn, lo, ss, asc tests all pass

**Comment by bnevis-i** [comment]
Pushed new commit to update to 2.0.0rc2.  Plan to sit on this until 2.0 official is released, or go back to a prior 1.x version if we hit code freeze before 2.0 is out...

**Comment by tingyuz** [comment]
Just heads up I am getting such error while trying 2.0.0-rc2 on arm64 environment. This behavior is different from x86 FYI ( see my comments above):

level=INFO ts=2020-01-21T21:02:57.452707936Z app=edgex-security-proxy-setup source=requestor.go:47 msg="successfully loaded rootCA certificate."
level=ERROR ts=2020-01-21T21:02:58.256049413Z app=edgex-security-proxy-setup source=init.go:61 msg="the status of service on http://kong:8001 is unknown, the initialization is terminated"

**Comment by tingyuz** [comment]
Dumping some logs from KONG here for future reference:

/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found
+ exec /consul/scripts/health kong-migrations
passing
/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found
+ exec /consul/scripts/health kong-migrations
passing
/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found
+ exec /consul/scripts/health kong-migrations
passing
/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found
+ exec /consul/scripts/health kong-migrations
passing
/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found
+ exec /consul/scripts/health kong-migrations
passing
/usr/local/openresty/nginx/sbin/nginx: 2: /usr/local/openresty/nginx/sbin/nginx: X��: not found
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx: Syntax error: Unterminated quoted string
/usr/local/openresty/nginx/sbin/nginx: 1: /usr/local/openresty/nginx/sbin/nginx:ELF: not found

**Comment by bnevis-i** [comment]
Turns out that I didn't git add on my repush and only changed the commit message.  I re-tested arm64 with rc2 and the issue with rc1 seems resolved.

**Comment by bnevis-i** [comment]
Flagged as on-hold until we get closer to code freeze.

**Comment by bnevis-i** [comment]
Updated to 2.0.0 (release version) and removed hold.

**Comment by tingyuz** [comment]
Verified kong2.0 on both x86 and arm, both working with master branch.

File: releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml [hunk 1]
```diff
@@ -166,7 +166,7 @@ services:
         - 'POSTGRES_USER=kong'
 
   kong-migrations:
-    image: kong:1.3.0-ubuntu
+    image: kong:2.0.0-ubuntu
```
Comment (hutchic) [inline releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml]: IMO set sensible defaults and let environment variables override. 

Would benefit other repositories such as `blackbox-testing` while keeping backwards compatibility
```suggestion
    image: kong:${KONG_VERSION:-2.0.1}-ubuntu
```

**Comment by bnevis-i** [comment]
@hutchic 

I had no idea docker-compose supported the full shell variable syntax.  Implemented suggestion.

**Pull request merged**